java/src/main/java/ai/onnxruntime/OrtSession.java

```diff
@@ -386,19 +383,43 @@ public String endProfiling() throws OrtException {
 
   @Override
   public String toString() {
-    return super.toString() + "(numInputs=" + numInputs + ",numOutputs=" + numOutputs + ")";
+    return super.toString()
+        + "(numInputs="
+        + getNumInputs()
+        + ",numOutputs="
+        + getNumOutputs()
+        + ")";
+  }
+
+  /**
+   * Terminate running model evaluations.
+   *
+   * <p>This is typically used in multi-threaded environments, where session management and model
+   * evaluation are occurring in different threads. This method can be used prior to calling {@link
+   * #close()} to prevent the normal behavior of blocking until pending model evaluations are
+   * completed.
+   *
+   * <p>Note: This will only terminate calls to {@link #run(Map)} and {@link #run(Map,Set)}. It will
```

Comment:
I had envisioned that the OrtSession would track all the currently executing runs and then call terminate on all the RunOptions that are there. Poke them into a HashSet on the way through and remove them on the way out.

Reply:
Yeah that makes more sense I'll update it